include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+value and N factors of the form `Tn(x'_i)`, where `x'_i` is the
+normalised value of the i'th input axis value.
+
+The forward and inverse transformations are defined independantly
+by separate sets of coefficients, supplied when the ChebyMap is
+created. If no coefficients are supplied to define the inverse
+transformation, the polyTran method can instead be used to create an
+inverse transformation. The inverse transformation so generated
+will be a Chebyshev polynomial with coefficients chosen to minimise
+the residuals left by a round trip (forward transformation followed
+by inverse transformation).
+
+### Attributes
+
+All those of Mapping. In addition, the forward and inverse bounds can be retrieved using getDomain
+
+Strictly speaking, ChebyMap it has all the attributes of PolyMap, but the only attributes PolyMap adds
+to Mapping are iterative inverse parameters and those are ignored by ChebyMap because it does not (yet)
+support an iterative inverse.
+*/
+class ChebyMap : public Mapping {
+    friend class Object;
+
+public:
+    /**
+    Construct a @ref ChebyMap with a specified forward and/or inverse transforms.
+
+    The two sets of coefficients are independent of each other: the inverse transform
+    need not undo the forward transform.
+
+    @param[in] coeff_f  A `(2 + nin) x ncoeff_f` @ref ChebyMap_CoefficientMatrices "matrix of coefficients"
+        describing the forward transformation. If `coeff_f` is empty then no forward transformation
+        is provided.
+    @param[in] coeff_i  A (2 + nout) x ncoeff_i` matrix of coefficients
+        for the inverse transformation. If coeff_i is empty then no inverse transformation is provided
+        unless you specify suitable options to request an iterative inverse; see the
+        @ref ChebyMap(ndarray::Array<double, 2, 2> const &, int, std::vector<double> const &,
+        std::string const &) "other constructor" for details.
+    @param[in] lbnd_f  Lower bounds for input data; one element per input axis
+    @param[in] ubnd_f  Upper bounds for input data; one element per input axis
+    @param[in] lbnd_i  Lower bounds for output data; one element per output axis
+    @param[in] ubnd_i  Upper bounds for output data; one element per output axis
+    @param[in] options  Comma-separated list of attribute assignments.
+
+    If a transform is not specified then the corresponding bounds are ignored (not even length-checked)
+    and can be empty. For example iff `coeff_f` is empty then `lbnd_f` and `ubnd_f` are ignored.
+
+    @throw std::invalid_argument if neither transform is specified (coeff_f and coeff_i are both empty).
+    @throw std::invalid_argument if the forward transform is specified (coeff_f is not empty)
+        and lbnd_f or ubnd_f do not have nin elements.
+    @throw std::invalid_argument if the inverse transform is specified (coeff_i is not empty)
+        and lbnd_i or ubnd_i do not have nout elements.
```

Comment:
I want to emphasize that the test only occurs if the transform is specified. I did try combining conditions for `_f` and `_i` into a single `@throw`, but was not satisfied with the results. I could change `nin elements` to `the correct number of elements` but it's longer.